src/ReverseProxy.Core/Service/Proxy/HttpProxy.cs

```diff
@@ -337,6 +337,12 @@ private void CopyHeadersToUpstream(IHeaderDictionary source, HttpRequestMessage
         {
             foreach (var header in source)
             {
+                var headerValueCount = header.Value.Count;
+                if (headerValueCount == 0)
+                {
+                    continue;
+                }
+
                 if (header.Key.Length > 0 && header.Key[0] == ':')
```

Comment:
Not related to your change, but this line should have a comment.
```suggestion
                // Filter out HTTP/2 pseudo headers like ":method" and ":path", those go into other fields.
                if (header.Key.Length > 0 && header.Key[0] == ':')
```